player/audio.c

```diff
@@ -182,6 +182,19 @@ void audio_update_volume(struct MPContext *mpctx)
     ao_set_gain(ao_c->ao, gain);
 }
 
+// Called when opts->ao_volume were changed.
+void audio_update_ao_volume(struct MPContext *mpctx)
+{
+    struct MPOpts *opts = mpctx->opts;
+    struct ao *ao = mpctx->ao;
+    float vol = opts->ao_volume;
+    if (!ao || !vol)
```

Comment:
So we cannot set 0 volume?

Reply:
True, I added this so that if no value was added it wouldn't override what was already set by the system, but forgot about the case when volume is 0. I can fix it by changing the verification to `vol < 0` and set the default value of opts->ao_volume to -1.